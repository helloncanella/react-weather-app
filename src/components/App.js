import React from 'react'
import { graphql } from 'react-apollo'

import { query } from '../data'

export const App = ({ data }) => {
  const { error, getWeather, loading } = data
  if (loading) {
    return <h1>Loading...</h1>
  } else if (error) {
    return <h1>Error...</h1>
  }
  const { current, location } = getWeather
  const { base, standard } = current
  const { city, country, currentTime, region } = location
  const header = `${city}, ${region}, ${country}`
  return (
    <div>
      <h1>Hello World</h1>
      <h2>{header}</h2>
      <h2>{currentTime}</h2>
      <div>
        <h2>
          Currently: {base.condition} & {standard.temp}F
        </h2>
        <h4>Feels like: {standard.feelsLike}F</h4>
        <div>
          <h3>Change of rain: {standard.precip}%</h3>
          <h3>
            Wind: {base.windCard} {standard.wind}mph
          </h3>
          <h3>Pressure: {standard.pres}in</h3>
          <h3>Visibility: {standard.vis}mi</h3>
        </div>
      </div>
    </div>
  )
}

export default graphql(query)(App)
