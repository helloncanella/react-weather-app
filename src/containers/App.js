import React, { Component } from 'react'

import { Landing, Layout, Logo, Search } from '../components'
import DataContainer from './DataContainer'

class App extends Component {
  state = {
    location: '',
    query: '',
    submitted: false
  }
  handleOnChange = e => {
    const query = e.target.value
    this.setState(state => ({ ...state, query }))
  }
  handleOnClick = e => {
    this.setState(state => ({ ...state, location: '', submitted: false }))
  }
  handleOnSubmit = e => {
    e.preventDefault()
    /**
     * NOTE
     * Must persist the query to `location`
     * since resetting `query` will cause graphql
     * to receive no `variable`.
     */
    this.setState(state => ({
      ...state,
      location: state.query,
      query: '',
      submitted: true
    }))
  }
  renderLanding = () => <Landing />
  renderSearch = () => (
    <Search
      onChange={this.handleOnChange}
      onSubmit={this.handleOnSubmit}
      query={this.state.query}
    />
  )
  renderWeather = () => (
    <DataContainer reset={this.handleOnClick} userQuery={this.state.location} />
  )
  render() {
    const { submitted } = this.state
    return (
      <Layout>
        {submitted ? this.renderWeather() : this.renderLanding()}
        {submitted ? null : this.renderSearch()}
        {submitted ? null : <Logo />}
      </Layout>
    )
  }
}

export default App
