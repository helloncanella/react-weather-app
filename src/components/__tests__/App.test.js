import React from 'react'
import { shallow } from 'enzyme'

import App from '../App'

describe('Stupid Test', () => {
  test('should behave...', () => {
    const wrapper = shallow(<App />)
    expect(wrapper.find('h1').text()).toEqual('Hello World')
    expect(wrapper).toMatchSnapshot()
  })
})
