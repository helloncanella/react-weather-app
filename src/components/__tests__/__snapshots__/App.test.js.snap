// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stupid Test should behave... 1`] = `
<h1>
  Hello World
</h1>
`;
